cmd/argo/lint/formatter_pretty_test.go

```diff
@@ -1,3 +1,5 @@
+//go:build !windows
```

Comment:
I think some comments would be helpful to add to these directives (in each file they were added) about why these are skipped for Windows tests.
Especially as the directives are used for testing, not building, which can be confusing if you just read the line and did not know the context

Reply:
I totally agree - I wasn't entirely happy with it. I'd much rather skip it just for windows tests but IIRC there's no other way than the approach I've taken for other tests - doing `t.Skip()` on windows OS. Not sure what's better?

Reply:
2 separate pieces there:

1. If we were to leave this in, a comment next to the directive would still be helpful. Right now there is no
2. To skip an entire file, you can use [`TestMain`](https://pkg.go.dev/testing#hdr-Main) with a `t.Skip` in it. That would be preferable and much more explicit, IMO. An example can be found in [this SO Answer](https://stackoverflow.com/a/64057110/3431180)

Reply:
makes sense :+1:

Reply:
see #12071